homeassistant/components/switch/tesla.py

```diff
@@ -0,0 +1,58 @@
+"""
+Support for Tesla charger switch.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.tesla/
+"""
+import logging
+
+from homeassistant.components.switch import ENTITY_ID_FORMAT, SwitchDevice
+from homeassistant.components.tesla import DOMAIN as TESLA_DOMAIN, TeslaDevice
+from homeassistant.const import STATE_ON, STATE_OFF
+
+_LOGGER = logging.getLogger(__name__)
+_VALID_STATES = [STATE_ON, STATE_OFF]
+
+DEPENDENCIES = ['tesla']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Tesla switch platform."""
+    devices = [ChargerSwitch(device, hass.data[TESLA_DOMAIN]['controller'])
+               for device in hass.data[TESLA_DOMAIN]['devices']['switch']]
+    add_devices(devices, True)
+
+
+class ChargerSwitch(TeslaDevice, SwitchDevice):
+    """Representation of a Tesla charger switch."""
+
+    def __init__(self, tesla_device, controller):
+        """Initialisation of the switch."""
+        self._state = None
+        super().__init__(tesla_device, controller)
+        self._name = self.tesla_device.name
+        self.entity_id = ENTITY_ID_FORMAT.format(self.tesla_id)
```

Comment:
That is not needed...

Reply:
Can we discuss that in discord? There is some questions from my side.

Reply:
This is optional. If you have a lot of Teslas with overlapping names, having it using the tesla ID is a good idea.

Reply:
So, Tesla component allow you to integrate a few Tesla's. It is why I'm using VIN number as a part of the entity_id. 
Could you please confirm if I can continue to use assignment of entity_id in my code?

Reply:
Yes, you can keep using the vin.